File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def middleNode(self, head: Optional[ListNode]) -> Optional[ListNode]:
        target = 1
        temp = head
        while temp.next:
            temp = temp.next
            target += 1
        else:
            target //= 2
        
        temp2= head
        while target:
            temp2 = temp2.next
            target -= 1
        else:
            return temp2

File: 0876-middle-of-the-linked-list/0876-middle-of-the-linked-list.dart
/**
 * Definition for singly-linked list.
 * class ListNode {
 *   int val;
 *   ListNode? next;
 *   ListNode([this.val = 0, this.next]);
 * }
 */
class Solution {
  ListNode? middleNode(ListNode? head) {
    int length = 1;
    var pointer = head;
    while (pointer?.next != null) {
        length += 1;
        pointer = pointer?.next;
    }
    int target = length ~/2;
    var answer = head;
    for (int i = 0;i < target;i++) {
        answer = answer?.next;
    }
    return answer;
  }
}